src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -11120,6 +11120,67 @@ instruct MoveL2D_reg_reg(regD dst, rRegL src) %{
   ins_pipe( pipe_slow );
 %}
 
+instruct FloatClassCheck_reg_reg_vfpclass(rRegI dst, regF src, kReg ktmp, rFlagsReg cr)
+%{
+  predicate(VM_Version::supports_avx512dq());
+  match(Set dst (IsInfiniteF src));
+  match(Set dst (IsNaNF src));
+  match(Set dst (IsFiniteF src));
+  effect(TEMP ktmp, KILL cr);
+  format %{ "float_class_check $dst, $src" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    __ float_class_check_vfp(opcode, $dst$$Register, $src$$XMMRegister, $ktmp$$KRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct FloatClassCheck_reg_reg(rRegI dst, regF src, rRegI tmp, rFlagsReg cr)
+%{
+  predicate(!VM_Version::supports_avx512dq());
+  match(Set dst (IsInfiniteF src));
+  match(Set dst (IsNaNF src));
+  match(Set dst (IsFiniteF src));
+  effect(TEMP tmp, KILL cr);
+  format %{ "float_class_check $dst, $src" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    __ float_class_check(opcode, $dst$$Register, $src$$XMMRegister, $tmp$$Register);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct DoubleClassCheck_reg_reg_vfpclass(rRegI dst, regD src, kReg ktmp, rFlagsReg cr)
+%{
+  predicate(VM_Version::supports_avx512dq());
+  match(Set dst (IsInfiniteD src));
+  match(Set dst (IsNaND src));
+  match(Set dst (IsFiniteD src));
+  effect(TEMP ktmp, KILL cr);
+  format %{ "double_class_check $dst, $src" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+      __ double_class_check_vfp(opcode, $dst$$Register, $src$$XMMRegister, $ktmp$$KRegister);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct DoubleClassCheck_reg_reg(rRegI dst, regD src, rRegL tmp, rRegL tmp1, rFlagsReg cr)
+%{
+  predicate(!VM_Version::supports_avx512dq());
+  match(Set dst (IsInfiniteD src));
+  match(Set dst (IsNaND src));
+  match(Set dst (IsFiniteD src));
+  effect(TEMP tmp, TEMP tmp1, KILL cr);
+  format %{ "double_class_check $dst, $src" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    __ double_class_check(opcode, $src$$XMMRegister, $dst$$Register, $tmp$$Register,
+                $tmp1$$Register);
+  %}
+  ins_pipe(pipe_slow);
+%}
+
```

Comment:
Should this be guarded by LP64 ? since associated macro assembly routines have that check ?

Also match_rule_supported vector should prevent intrinsifying these for 32 bit platform if you do not plan to handle them.